nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -367,4 +386,74 @@ with lib;
       description = "Declarative location config";
     };
   };
+
+  config = let
+   inherit (global-config.services.nginx) package;
+  in {
+    warnings = optional config.enableSSL
+      ''
+        config.services.nginx.virtualHosts."${name}".enableSSL is deprecated,
+        use config.services.nginx.virtualHosts."${name}".onlySSL instead.
+      '';
+
+    assertions = [
+      {
+        assertion = all (l: l.root == null || l.alias == null) (attrValues config.locations);
+        message = "Only one of nginx root or alias can be specified on a location.";
```

Comment:
This evaluates to ``Only one of nginx root or alias can be specified on a location.`` and we should add a location here, too.